compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -1973,6 +1992,13 @@ private BType checkIndexAccessExpr(BLangIndexBasedAccess indexBasedAccessExpr, B
                 checkExpr(indexExpr, this.env);
                 actualType = symTable.xmlType;
                 break;
+            case TypeTags.TUPLE:
+                indexExprType = checkIndexExprForTupleFieldAccess(indexExpr);
+                if (indexExprType.tag == TypeTags.INT) {
+                    int indexValue = ((Long) ((BLangLiteral) indexExpr).value).intValue();
```

Comment:
can't we support expressions here?(with a runtime check?)

Reply:
Due to technical complexity of identifying the correct type within a tuple, we will limit this to integer literal